polkadot/node/core/approval-voting/src/criteria.rs

```diff
@@ -374,6 +475,84 @@ fn compute_relay_vrf_modulo_assignments(
 	}
 }
 
+fn assigned_cores_transcript(core_bitfield: &CoreBitfield) -> Transcript {
+	let mut t = Transcript::new(approval_types::v2::ASSIGNED_CORE_CONTEXT);
+	core_bitfield.using_encoded(|s| t.append_message(b"cores", s));
+	t
+}
+
+fn compute_relay_vrf_modulo_assignments_v2(
+	assignments_key: &schnorrkel::Keypair,
+	validator_index: ValidatorIndex,
+	config: &Config,
+	relay_vrf_story: RelayVRFStory,
+	leaving_cores: Vec<(CandidateHash, CoreIndex)>,
+	assignments: &mut HashMap<CoreIndex, OurAssignment>,
+) {
+	let mut assigned_cores = Vec::new();
+	let leaving_cores = leaving_cores.iter().map(|(_, core)| core).collect::<Vec<_>>();
+
+	let maybe_assignment = {
+		let assigned_cores = &mut assigned_cores;
+		assignments_key.vrf_sign_extra_after_check(
+			relay_vrf_modulo_transcript_v2(relay_vrf_story.clone()),
+			|vrf_in_out| {
+				*assigned_cores = relay_vrf_modulo_cores(
+					&vrf_in_out,
+					config.relay_vrf_modulo_samples,
+					config.n_cores,
+				)
+				.into_iter()
+				.filter(|core| leaving_cores.contains(&core))
+				.collect::<Vec<CoreIndex>>();
```

Comment:
We should not produce assignments for cores we backed.

Reply:
We don't leaving_cores does not include the backed cores see how it is created:
```
	let leaving_cores = leaving_cores
		.into_iter()
		.filter(|(_, _, g)| !is_in_backing_group(&config.validator_groups, index, *g))
		.map(|(c_hash, core, _)| (c_hash, core))
		.collect::<Vec<_>>();
```

Reply:
nvm, this is already filtered in `leaving_cores`